### ISSUE TITLE
Implementation of convolution of PDF

### ISSUE THREAD

**Issue by abhijitm08** [zfit/zfit#137]
Addition of convolution feature of two PDFs.

I have implemented here using the tf.nn.conv1d for 1D (which internally uses tf.nn.conv2d with reshaping tensors): https://drive.google.com/file/d/14wE2gFjgSVjtSDypOaaKPTshxuek9qDF/view?usp=sharing 

However need to generalise this to multidimensional case.

**Comment by abhijitm08** [comment]
Hey, Here is the convolution implemented for 1d and 2d cases (easily extendable to multidimensional case) when the function and kernel are discrete. Can you please put into zfit? If function and kernel are already evaluated and discrete then we just ought to use inbuilt TF functions. Also for the continuous case, I can't quite see how Fourier transform is computationally inexpensive, but this is something to thing about. 

https://colab.research.google.com/drive/1_a11xrRO_Sb6YmI1gf1Y9-wtISj0N-97

**Comment by mayou36** [comment]
I've implemented something in https://github.com/zfit/zfit-physics/pull/3
It supports n dim convolutions with mc integration

It's available in the namespace `zphys.unstable.pdf.ConvPDF`, be aware that API etc may change (let me know for ideas of improvements)

**Comment by mayou36** [comment]
There is now an other convolution pdf in `zfit.pdf.FFTConvPDFV1` available
